Add support for liquid in SVG

6twenty opened the pull request:
- [x] I have added or updated the specs/tests.
- [x] I have verified that the specs/tests pass on my computer.
- [x] I have not attempted to bump, or alter versions.
- [ ] This is a documentation change.
- [x] This is a source change.

Currently assets can be processed by Liquid by appending or prepending `.liquid` to the extension, however this only applies to javascript and css (including sass and coffeescript). It would be useful to be able to include SVG files to this list, because SVG is capable of referencing other files - for example using `<image xlink:href="..."></image>`. Using liquid would enable referencing the digest path of a compiled asset using `<image xlink:href="{% asset path/to/asset @path %}"></image>`.

This PR adds a new `image/liquid+svg+xml` entry as well as a change to `Utils#strip_secondary_content_type` in order to support it.

envygeeks commented:
Can you update the readme to note this please?

6twenty commented:
It's not really clear how to add this to the readme - it mentions being able to use `.liquid.ext` or `.ext.liquid`, but doesn't specifically mention which extensions are supported. Would you be happy with a list along the lines of this:

-----

### `.liquid.ext`
### `.ext.liquid`

#### Supported file types:

- .css
- .sass
- .scss
- .js
- .es6
- .coffee
- .svg

envygeeks commented:
I guess I don't carry full proper documentation on the supported extensions, I thought I did.  Indeed I would be extremely happy if you just add a list so it's more clear to people! Otherwise I can do it, I'm unblocking your pull on my end.

6twenty commented:
@envygeeks I've updated the readme - I replaced the small sass code block under that heading since it didn't seem related to the `.liquid` file extension and uses the same syntax as the code block above, under the **`.sass`, `.scss`** heading. So this section of the readme is now:

-----

## Liquid

We support liquid arguments for tag values (but not tag keys), and we also support Liquid pre-processing (with your Jekyll context) of most files if they end with `.liquid`.  This will also give you access to our filters as well as their filters, and Jekyll's filters, and any tags that are globally available.

```liquid
{% asset '{{ site.bg_img }}' %}
{% asset '{{ site.bg_img }}' proxy:key='{{ value }}' %}
{% asset {{\ site.bg_img\ }} %}
```

### `.sass`, `.scss`

```scss
body {
  background-image: asset_url("'{{ site.bg_img }}'");
  background-image: asset_url("'{{ site.bg_img }}' proxy:key='{{ value }}'");
  background-image: asset_url("{{\ site.bg_img\ }}");
}
```

### `.liquid.ext`
### `.ext.liquid`

#### Supported file types:

- .css
- .sass
- .scss
- .js
- .es6
- .coffee
- .svg

envygeeks commented:
Thanks ❤️
